### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by sfcsumb**
Fixed missing query in AssignmentRepository.java for testing and renamed function to clarify it is to find an assignment.

### CODE REVIEW

File: cst438_gradebook/src/main/java/com/cst438/domain/AssignmentRepository.java [hunk 1]
```diff
@@ -10,4 +10,7 @@ public interface AssignmentRepository extends CrudRepository <Assignment, Intege
 
 	@Query("select a from Assignment a where a.needsGrading=1 and a.dueDate < current_date and a.course.instructor= :email order by a.id")
 	List<Assignment> findNeedGradingByEmail(@Param("email") String email);
+
+	@Query("select a from Assignment where course_id= ? and a.name= name", nativeQuery=true)
+	List<Assignment> findAssignmentByCourseIdAndName(int id, String name);
```
Comment (denizozgur) [inline cst438_gradebook/src/main/java/com/cst438/domain/AssignmentRepository.java:15]:
```suggestion

	@Query("select a from Assignment a where a.course.course_id= :id and a.name= :name")
	List<Assignment> findAssignmentByCourseIdAndName(@Param("id") int id,@Param("name") String name);
```

File: cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java [hunk 2]
```diff
@@ -78,15 +78,15 @@ public void testCreateAssignment() throws Exception {
 			driver.findElement(By.xpath("//input[@name='handleSubmit']")).click();
 			Thread.sleep(SLEEP_DURATION);
 			
-			Optional<Assignment> a = assignmentRepository.findByCourseIdAndName(TEST_COURSE_ID, TEST_ASSIGNMENT_NAME);
+			Optional<Assignment> a = assignmentRepository.findAssignmentByCourseIdAndName(TEST_COURSE_ID, TEST_ASSIGNMENT_NAME);
```
Comment (denizozgur) [inline cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java:81]: Doesn't return Optional<Assignment> anymore 

```suggestion
			List<Assignment> a = assignmentRepository.findAssignmentByCourseIdAndName(Integer.parseInt(TEST_COURSE_ID), TEST_ASSIGNMENT_NAME);
```

File: cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java [hunk 3]
```diff
@@ -78,15 +78,15 @@ public void testCreateAssignment() throws Exception {
...
 			if (a != null) {
 				assignmentRepository.delete(a);
```
Comment (denizozgur) [inline cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java:83]: delete requires a single entity below will be the first item of the list if it's populated
```suggestion
				assignmentRepository.delete(a.get(0));
```

File: cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java [hunk 4]
```diff
@@ -78,15 +78,15 @@ public void testCreateAssignment() throws Exception {
...
 			}
 			
 		} catch(Exception e) {
 			throw e;
 		} finally {
-			Optional<Assignment> a = assignmentRepository.findByCourseIdAndName(TEST_COURSE_ID, TEST_ASSIGNMENT_NAME);
+			Optional<Assignment> a = assignmentRepository.findAssignmentByCourseIdAndName(TEST_COURSE_ID, TEST_ASSIGNMENT_NAME);
 			if (a != null) {
 				assignmentRepository.delete(a);
```
Comment (denizozgur) [inline cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java:91]: same problem as line 83

**Review by denizozgur — CHANGES_REQUESTED** [review]
I can't compile the code without these changes

File: cst438_gradebook/src/main/java/com/cst438/domain/AssignmentRepository.java (hunk 1 above)
Comment (sfcsumb) [reply cst438_gradebook/src/main/java/com/cst438/domain/AssignmentRepository.java:15]: Sorry, you are correct.

File: cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java (hunk 4 above)
Comment (sfcsumb) [reply cst438_gradebook/src/test/java/com/cst438/EndToEndTestAssignments.java]: Fixed

**Review by denizozgur — APPROVED** [review]

**Pull request merged**